src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -819,13 +815,22 @@ private static Class<?> loadModuleMainClass(String what) {
      */
     private static Class<?> loadMainClass(int mode, String what) {
         // get the class name
-        String cn;
+        String cn = null;
+        // In LM_JAR mode, put the underlying file in the JarFile/ZipFile cache.
+        // This will avoid needing to re-parse the manifest when the JAR file
+        // is opened on the class path, triggered by Class.forName below.
```

Comment:
Nice improvement to have. would it be ok to pad out this comment a bit more ? I found it difficult to understand the specifics. Perhaps :

```
		// In LM_JAR mode, keep the underlying file open to retain it in 
		// the ZipFile HashMap cache. This will avoid needing to re-parse
                // the central directory when the file is opened on the class path,
		// triggered by Class.forName below.
```

Reply:
> Nice improvement to have. would it be ok to pad out this comment a bit more ? I found it difficult to understand the specifics. Perhaps :
> 
> ```
> 		// In LM_JAR mode, keep the underlying file open to retain it in 
> 		// the ZipFile HashMap cache. This will avoid needing to re-parse
>                 // the central directory when the file is opened on the class path,
> 		// triggered by Class.forName below.
> ```

No issue with expanding the comment but I think leave it as "ZipFile cache" as it will quickly bit rot, and of course the Launcher code doesn't care which collection is used internally in the ZipFile code.

Reply:
I'll expand the comment as suggest, keeping the "JarFile/ZipFile cache" phrase.